apps/server/src/modules/room/domain/service/room-arrangement.service.ts

```diff
@@ -0,0 +1,76 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain/types';
+import { RoomArrangementRepo } from '../../repo';
+import { RoomArrangementItem } from '../type';
+import { RoomService } from './room.service';
+
+@Injectable()
+export class RoomArrangementService {
+	constructor(private readonly roomService: RoomService, private readonly roomArrangementRepo: RoomArrangementRepo) {}
+
+	public async sortRoomIdsByUserArrangement(userId: EntityId, roomIds: EntityId[]): Promise<EntityId[]> {
+		const arrangementExists = await this.roomArrangementRepo.hasArrangementForUserId(userId);
```

Comment:
I don't gave much thought, but just wonder if calling an upsert in repo would be possible, and so having less code

Reply:
The logic is quite different. So I decided to not use an upsert. Doesn't mean that's not possible though. I would leave it like it is for now.